vnext/Microsoft.ReactNative/ReactApplication.cpp

```diff
@@ -96,7 +98,12 @@ void ReactApplication::JavaScriptBundleFile(hstring const &value) noexcept {
 void ReactApplication::OnActivated(Windows::ApplicationModel::Activation::IActivatedEventArgs const &e) {
   if (e.Kind() == Windows::ApplicationModel::Activation::ActivationKind::Protocol) {
     auto protocolActivatedEventArgs{e.as<Windows::ApplicationModel::Activation::ProtocolActivatedEventArgs>()};
-    ::Microsoft::ReactNative::LinkingManagerModule::OpenUri(protocolActivatedEventArgs.Uri());
+    const auto uri = protocolActivatedEventArgs.Uri();
+    const auto properties = InstanceSettings().Properties();
+    if (LinkingHelper::InitialUrl(properties).empty()) {
+      LinkingHelper::SetInitialUrl(properties, uri.AbsoluteUri());
+    }
+    ::Microsoft::ReactNative::LinkingManager::OpenUri(uri);
```

Comment:
We should put OpenUri on the public Linking class.  -- If ReactApplication is using anything internal, its an indication of some missing exposure.  (I personally think most of what is in ReactApplication should actually be part of the template rather than a built in class)